features/nvstore/source/nvstore_shared_lock.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2016 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#include "nvstore_shared_lock.h"
+
+#include "mbed.h"
+
+#include "mbed_critical.h"
+#include "rtos/thread.h"
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifndef NVSTORE_THREAD_SAFE
```

Comment:
Why is this macro needed? Is it explained anywhere? Why would we want to disable this?

Reply:
Originally this was there in the client code in order to allow users to reduce code size, if it is clear that no thread safety is required (for instance if there's only one thread). This will be removed altogether, as well as the NVSTORE_PROBE_ONLY define (reducing code size even more aggressively.